### PR TITLE
add shrinkwrap. remove fsevents and move it into optionalDependencies…

### PR DESCRIPTION
**Pull request by Jonahss**
…. see https://github.com/npm/npm/issues/2679

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -3,7 +3,21 @@
...
-    fs = require('fs');
+    fs = require('fs'),
+    shrinkwrap = require('./npm-shrinkwrap.json');
+
+// remove 'fsevents' from shrinkwrap, since it causes errors on non-Mac hosts
+// see https://github.com/npm/npm/issues/2679
+gulp.task('fixShrinkwrap', function (done) {
+  delete shrinkwrap.dependencies.fsevents;
+  var shrinkwrapString = JSON.stringify(shrinkwrap, null, '  ') + '\n';
+  fs.writeFile('./npm-shrinkwrap.json', shrinkwrapString, function (err) {
```
Comment (imurchie) [inline gulpfile.js]: Can't you just pass in `done` here?

Comment (Jonahss) [inline gulpfile.js]: heh, jlipps saw the same thing.

**Comment by imurchie** [comment]
LGTM.

File: test/shrinkwrap-specs.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    require('../../npm-shrinkwrap.json');
+  });
+
+  it('shrinkwrap should not include fsevents', async () => {
+    // fsevents is an optional dep that only works on Mac.
+    // if it's in shrinkwrap, non-Mac hosts won't be able to install appium
+    let shrinkwrap = require('../../npm-shrinkwrap.json');
+    should.exist(shrinkwrap.dependencies);
+    _.values(shrinkwrap.dependencies).length.should.be.above(10);
+    should.not.exist(shrinkwrap.dependencies.fsevents);
```
Comment (imurchie) [inline test/shrinkwrap-specs.js]: is there a way to pull `fsevents` out of `package.json`'s optional dependencies? Rather than hard coding it?

**Pull request merged**